client/pd_service_discovery.go

```diff
@@ -78,25 +79,23 @@ type BaseClient interface {
 	// in a quorum-based cluster or any primary/secondary in a primary/secondary configured cluster
 	// is changed.
 	AddServiceAddrsSwitchedCallback(callbacks ...func())
+}
+
+type tsoAllocServingAddrUpdatedFunc func(map[string]string) error
 
-	// TODO: Separate the following TSO related service discovery methods from the above methods.
-
-	// GetTSOAllocators returns {dc-location -> TSO allocator leader URL} connection map
-	GetTSOAllocators() *sync.Map
-	// GetTSOAllocatorServingAddrByDCLocation returns the tso allocator of the given dcLocation
-	GetTSOAllocatorServingAddrByDCLocation(dcLocation string) (string, bool)
-	// GetTSOAllocatorClientConnByDCLocation returns the tso allocator grpc client connection
-	// of the given dcLocation
-	GetTSOAllocatorClientConnByDCLocation(dcLocation string) (*grpc.ClientConn, string)
-	// AddTSOAllocatorServingAddrSwitchedCallback adds callbacks which will be called
-	// when any global/local tso allocator service endpoint is switched.
-	AddTSOAllocatorServingAddrSwitchedCallback(callbacks ...func())
+type tsoAllocatorEventSource interface {
+	// AddTSOAllocServingAddrsUpdatedCallback adds callbacks which will be called
+	// when the global/local tso allocator leader list is updated.
+	AddTSOAllocServingAddrsUpdatedCallback(callbacks ...tsoAllocServingAddrUpdatedFunc)
```

Comment:
How about making the name shorter?

Reply:
AddTSOAllocServingAddrsUpdatedCallback -> AddTSOServAddrsUpdatedCallback